### ISSUE TITLE
S3CacheTest.testPurge:330 is a flaky test

### ISSUE THREAD

**Issue by jcoyne** [cantaloupe-project/cantaloupe#678]
This test seems to fail randomly on CI:
```
cantaloupe_1  | [ERROR]   S3CacheTest.testPurge:330 expected: <false> but was: <true>
```

**Comment by DiegoPino** [comment]
@jcoyne yes, it might be that these S3 tests can't act on Cache being written or now, so it waits 1/2 of the upload time only. If the backend is slow it might not be enough. Wonder if we could use `Mockito` in the future (no backed - just mocking the classes) (but requires probably a lot of rewriting) or this helper library to help with actually waiting?
`https://github.com/awaitility/awaitility`

For Mockito: e.g https://github.com/aws/aws-sdk-java-v2/issues/3368 (the example test there is almost good, see comment of AWS folks down there)

**Comment by glenrobson** [comment]
Glen to have a look to see if testPurge is using the async.

**Comment by glenrobson** [comment]
Test is still flaky

**Comment by glenrobson** [comment]
I've gone through the failed tests since July and its not just the S3CacheTest.testPurge which is causing the issue. Figures for the occurane of the failed test. 

* 3 - S3CacheTest.testPurge:338 expected: <false> but was: <true>
* 5 - S3CacheTest>AbstractCacheTest.testPurgeInfos:476 expected: <false> but was: <true>
* 3 - S3CacheTest.testPurgeWithIdentifier:530->AbstractCacheTest.testPurgeWithIdentifier:382 expected: <false> but was: <true>
* 6 - S3CacheTest.testPurgeWithKeyPrefix:392 expected: <false> but was: <true>

 * 8 - JdbcCacheTest.testPurge:286->AbstractCacheTest.testPurge:322->AbstractCacheTest.assertExists:36
 * 4 - JdbcCacheTest>AbstractCacheTest.testPurgeInfos:469->AbstractCacheTest.assertExists:36 expected: not <null>

**Comment by glenrobson** [comment]
Diego to have a quick look to see if there are any junit annotations we could use to get a test to run a number of times if it fails.

**Comment by glenrobson** [comment]
Maybe have a look at this: https://github.com/artsok/rerunner-jupiter

**Comment by DiegoPino** [comment]
@glenrobson and there is also this one
https://junit-pioneer.org/docs/retrying-test/ 

Got it from this blogpost (https://glaforge.dev/posts/2024/09/01/a-retryable-junit-5-extension/) , see the "Update" at the end
